### PR TITLE
Refactor registry etcd delete tests

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref #12207

### CODE REVIEW

**Comment by wojtek-t** [comment]
Test failure is a flake:
Kubernetes e2e suite.Services should serve multiport endpoints from pods

@k8s-bot test this please

File: pkg/api/rest/resttest/resttest.go [hunk 1]
```diff
@@ -160,50 +167,45 @@ func (t *Tester) TestUpdate(valid runtime.Object, setFn SetFunc, setRVFn SetRVFu
...
 
-// Test graceful deletion.
-// TODO(wojtek-t): Change it to use AssignFunc instead.
-func (t *Tester) TestDeleteGraceful(createFn func() runtime.Object, expectedGrace int64, wasGracefulFn func() bool) {
-	t.TestDeleteGracefulHasDefault(createFn(), expectedGrace, wasGracefulFn)
-	t.TestDeleteGracefulWithValue(createFn(), expectedGrace, wasGracefulFn)
-	t.TestDeleteGracefulUsesZeroOnNil(createFn(), 0)
-	t.TestDeleteGracefulExtend(createFn(), expectedGrace, wasGracefulFn)
-	t.TestDeleteGracefulImmediate(createFn(), expectedGrace, wasGracefulFn)
+// Test grafully deleting an object.
```
Comment (nikhiljindal) [inline pkg/api/rest/resttest/resttest.go]: Did you mean gracefully?

**Comment by nikhiljindal** [comment]
LGTM, thanks!

File: pkg/api/rest/resttest/resttest.go (hunk 1 above)
Comment (wojtek-t) [inline pkg/api/rest/resttest/resttest.go]: Done.

**Comment by wojtek-t** [comment]
Thanks a ot for review @nikhiljindal !

**Pull request merged**